src/toolchain/mod.rs

```diff
@@ -17,3 +21,10 @@ macro_rules! builtin_templates {
         ]
     }
 }
+
+fn create_toolchain<T: BuildTemplate>(toolchain: &T, data: TemplateData) -> Result<()> {
```

Comment:
Good idea! Indeed, that section of code is repeated in all files in the toolchain.